service/interpreter/InternalChannel.go

```diff
@@ -23,6 +26,16 @@ func (i *InternalChannel) GetAllReceived() map[string][]*iwfidl.EncodedObject {
 	return i.receivedData
 }
 
+func (i *InternalChannel) GetInfos() map[string]iwfidl.ChannelInfo {
+	infos := map[string]iwfidl.ChannelInfo{}
+	for name, l := range i.receivedData {
+		infos[name] = iwfidl.ChannelInfo{
+			Size: ptr.Any(int32(len(l))),
```

Comment:
Size is the number of messages, not actual data size in bytes, right?

Reply:
Yeah, l is the list of all the messages. So len(l) is the number of the messages. 

`receivedData` is a map of list.